### PR TITLE
Feature/display-sd-card

### PR DESCRIPTION
**Pull request by KarlSchu**
Integration of SD Card (read/write)
Fix conflicts with TFT display

### CODE REVIEW

File: build/CMakeCache.txt [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+########################
+
+//Path to a program.
+CMAKE_ADDR2LINE:FILEPATH=/usr/bin/addr2line
+
+//Path to a program.
+CMAKE_AR:FILEPATH=/usr/bin/ar
+
+//No help, variable specified on the command line.
+CMAKE_BUILD_TYPE:STRING=Release
```
Comment (andreaskuster) [inline build/CMakeCache.txt:24]: Is this file in use? I guess not? -> remove

File: build/CMakeFiles/3.16.3/CMakeCXXCompiler.cmake [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+set(CMAKE_CXX98_COMPILE_FEATURES "cxx_std_98;cxx_template_template_parameters")
+set(CMAKE_CXX11_COMPILE_FEATURES "cxx_std_11;cxx_alias_templates;cxx_alignas;cxx_alignof;cxx_attributes;cxx_auto_type;cxx_constexpr;cxx_decltype;cxx_decltype_incomplete_return_types;cxx_default_function_template_args;cxx_defaulted_functions;cxx_defaulted_move_initializers;cxx_delegating_constructors;cxx_deleted_functions;cxx_enum_forward_declarations;cxx_explicit_conversions;cxx_extended_friend_declarations;cxx_extern_templates;cxx_final;cxx_func_identifier;cxx_generalized_initializers;cxx_inheriting_constructors;cxx_inline_namespaces;cxx_lambdas;cxx_local_type_template_args;cxx_long_long_type;cxx_noexcept;cxx_nonstatic_member_init;cxx_nullptr;cxx_override;cxx_range_for;cxx_raw_string_literals;cxx_reference_qualified_functions;cxx_right_angle_brackets;cxx_rvalue_references;cxx_sizeof_member;cxx_static_assert;cxx_strong_enums;cxx_thread_local;cxx_trailing_return_types;cxx_unicode_literals;cxx_uniform_initialization;cxx_unrestricted_unions;cxx_user_literals;cxx_variadic_macros;cxx_variadic_templates")
```
Comment (andreaskuster) [inline build/CMakeFiles/3.16.3/CMakeCXXCompiler.cmake:10]: Same here (should be removed)

File: lib/CmdHandler/CmdHandler.cpp [hunk 3]
```diff
@@ -140,7 +142,7 @@ void CmdHandler::task() {
         carState.MotorCurrent = floatValue;
         break;
       case '-':
-        carControl.adjust_paddles(10);
+        carControl.adjust_paddles(3);
```
Comment (andreaskuster) [inline lib/CmdHandler/CmdHandler.cpp:145]: What is this value about? Maybe a constant in a class/parameter?

File: lib/SDCard/SDCard.cpp [hunk 4]
```diff
@@ -4,85 +4,103 @@
...
+      Serial.print('\t');
+    }
+    Serial.print(entry.name());
+    if (entry.isDirectory()) {
+      Serial.println("/");
+      printDirectory(entry, numTabs + 1);
+    } else {
+      // files have sizes, directories do not
+      Serial.print("\t\t");
+      Serial.println(entry.size(), DEC);
```
Comment (andreaskuster) [inline lib/SDCard/SDCard.cpp:49]: Everywhere else you changed to Cpp-Style `cout << `

File: src/main.cpp [hunk 5]
```diff
@@ -205,28 +195,28 @@ void app_main(void) {
     esp32time.setTime(now.Second(), now.Minute(), now.Hour(), now.Day(), now.Month(), now.Year());
     engineerDisplay.print("[v] " + rtc.getName() + " initialized, time in esp32 updated.\n");
   }
-  if (SD_ON) {
-    xTaskCreate(&write_sdcard_demo_task, "write_sdcard_demo_task", CONFIG_ESP_SYSTEM_EVENT_TASK_STACK_SIZE, NULL, 5, NULL);
-  }
+  // if (SD_ON) {
+  //   xTaskCreate(&write_sdcard_demo_task, "write_sdcard_demo_task", CONFIG_ESP_SYSTEM_EVENT_TASK_STACK_SIZE, NULL, 5, NULL);
+  // }
   if (INT_ON) {
```
Comment (andreaskuster) [inline src/main.cpp:201]: Why is this commented out? Shouldn't the sd card work now?

**Review by andreaskuster — CHANGES_REQUESTED** [review]
Well, obviously the formatting has to done and the build should pass..

The are a lot of files from the `build` directory that should probaly not be in here..

**Comment by KarlSchu** [comment]
> Why is this commented out? Shouldn't the sd card work now? 

Sd Card works, does not need a DemoTask. Will remove it.

File: src/main.cpp (hunk 5 above)
Comment (KarlSchu) [reply src/main.cpp:201]: removed, card works noch demo task necessary

File: build/CMakeCache.txt (hunk 1 above)
Comment (KarlSchu) [reply build/CMakeCache.txt:24]: Do we need anything from build in git repo?
Comes in by accident.

File: build/CMakeFiles/3.16.3/CMakeCXXCompiler.cmake (hunk 2 above)
Comment (KarlSchu) [reply build/CMakeFiles/3.16.3/CMakeCXXCompiler.cmake]: removed

File: lib/CmdHandler/CmdHandler.cpp (hunk 3 above)
Comment (KarlSchu) [reply lib/CmdHandler/CmdHandler.cpp:145]: This is the time (about) waiting for paddle input at adjusting. Same in boot sequence but longer.
If you has a good place we can migrate it

File: lib/SDCard/SDCard.cpp (hunk 4 above)
Comment (KarlSchu) [reply lib/SDCard/SDCard.cpp:49]: done,

File: build/CMakeCache.txt (hunk 1 above)
Comment (andreaskuster) [reply build/CMakeCache.txt]: I do not think that we need anything. Might be a directory that we should put into the `.gitignore` file as well.

File: lib/CmdHandler/CmdHandler.cpp (hunk 3 above)
Comment (andreaskuster) [reply lib/CmdHandler/CmdHandler.cpp:145]: Well, we can surely let it here, but I suggest to add ad comment that this is the wait time.

File: build/CMakeCache.txt (hunk 1 above)
Comment (KarlSchu) [reply build/CMakeCache.txt]: ignored by .gitignore

**Comment by KarlSchu** [comment]
should be merged now

**Review by andreaskuster — APPROVED** [review]

**Pull request merged**